### PR TITLE
http-api: encode Color & Instant more nicely

### PR DESCRIPTION
**Pull request by abextm**
Follow up to #12973

### CODE REVIEW

**Review by Nightfirecat — APPROVED** [review]
LGTM

File: http-api/src/main/java/net/runelite/http/api/gson/InstantTypeAdapter.java [hunk 1]
```diff
@@ -60,6 +58,12 @@ public Instant read(JsonReader in) throws IOException
 			return null;
 		}
 
+		if (in.peek() == JsonToken.NUMBER)
+		{
+			long jsTime = in.nextLong();
+			return Instant.ofEpochSecond(jsTime / 1000, (jsTime % 1000) * NS_PER_MS);
```
Comment (Adam-) [inline http-api/src/main/java/net/runelite/http/api/gson/InstantTypeAdapter.java:64]:
```suggestion
			return Instant.ofEpochMilli(jsTime);
```

File: http-api/src/main/java/net/runelite/http/api/gson/InstantTypeAdapter.java [hunk 2]
```diff
@@ -43,12 +45,8 @@ public void write(JsonWriter out, Instant value) throws IOException
 			return;
 		}
 
-		out.beginObject()
-			.name("seconds")
-			.value(value.getEpochSecond())
-			.name("nanos")
-			.value(value.getNano())
-			.endObject();
+		long jsTime = (value.getEpochSecond() * 1000) + (value.getNano() / NS_PER_MS);
```
Comment (Adam-) [inline http-api/src/main/java/net/runelite/http/api/gson/InstantTypeAdapter.java:48]: You can also use `value.getLong(ChronoField.MILLI_OF_SECOND)` for getting the ms I think.

File: http-api/src/main/java/net/runelite/http/api/gson/InstantTypeAdapter.java [hunk 3]
```diff
@@ -34,6 +34,8 @@
 // Just add water!
 public class InstantTypeAdapter extends TypeAdapter<Instant>
 {
+	private static final int NS_PER_MS = 1_000_000;
```
Comment (Adam-) [inline http-api/src/main/java/net/runelite/http/api/gson/InstantTypeAdapter.java:37]: This can be removed after the below changes

**Pull request merged**